### PR TITLE
CI: Doxygen

### PR DESCRIPTION
**Pull request by ax3l**
## Summary

Run Doxygen and make sure it does not ~~throw warnings~~ *error while parsing* in CI. This is already run as such in downstream projects, so we need to fix our upstream errors as well.

## Checklist

The proposed changes:
- [x] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: .github/workflows/style/doxygen [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+#
+# Copyright 2020 Axel Huebl
+#
+# License: BSD-3-Clause-LBNL
+
+# search recursive inside a folder if a file contains tabs
```
Comment (WeiqunZhang) [inline .github/workflows/style/doxygen]: The comments need to be updated.

**Comment by ax3l** [comment]
A few more to fix (help wanted :) ), currently stuck at
```
/home/axel/src/amrex/Src/Base/AMReX_GpuReduce.H:510: error: no matching file member found for 
AMREX_GPU_DEVICE AMREX_FORCE_INLINE int amrex::Gpu::blockReduceLogicalAnd(int source)
Possible candidates:
 'AMREX_GPU_DEVICE AMREX_FORCE_INLINE int blockReduceLogicalAnd(int source) noexcept' at line 494 of file /home/axel/src/amrex/Src/Base/AMReX_GpuReduce.H
 (warning treated as error, aborting now)
```

**Comment by WeiqunZhang** [comment]
Maybe it's confused by that a function template and a non-template function have the same function signature.

File: Src/Base/AMReX_Random.cpp [hunk 2]
```diff
@@ -27,7 +27,7 @@ namespace amrex {
 
 #ifdef AMREX_USE_GPU
 namespace {
-void ResizeRandomSeed (amrex::ULong gpu_seed)
+void ResizeRandomSeed (ULong gpu_seed)
```
Comment (ax3l) [inline Src/Base/AMReX_Random.cpp]:
```suggestion
void ResizeRandomSeed (amrex::ULong gpu_seed)
```

File: Docs/Doxygen/doxygen.conf [hunk 3]
```diff
@@ -2037,7 +2037,15 @@ SEARCH_INCLUDES        = YES
 # preprocessor.
 # This tag requires that the tag SEARCH_INCLUDES is set to YES.
 
-INCLUDE_PATH           =
+INCLUDE_PATH           = ../../Src/Amr      \
```
Comment (WeiqunZhang) [inline Docs/Doxygen/doxygen.conf]: Could you tell me why need to do this now?  Should it also include all the directories recursively?

Comment (ax3l) [reply Docs/Doxygen/doxygen.conf]: Somehow it did not find some includes (already) - we just ignored it as warnings before it seems.

**Comment by ax3l** [comment]
@WeiqunZhang ready to merge - the potential overflow in Random seems to be pre-existing (do we want to fix the cast separately?)

**Comment by WeiqunZhang** [comment]
I think the overflow warning is false positive.  We are doing something like

**Comment by WeiqunZhang** [comment]
Accidentally clicked the wrong button.

File: Docs/Doxygen/doxygen.conf (hunk 3 above)
Comment (WeiqunZhang) [reply Docs/Doxygen/doxygen.conf]: > Note that the INCLUDE_PATH is not recursive, so the setting of [RECURSIVE](https://www.doxygen.nl/manual/config.html#cfg_recursive) has no effect here.

My point is we probably need to include ../../Src/Base/Parser etc. as well, if this is needed.

```
../../Src/Base \
../../Src/Base/Parser \
../../Src/SDC \
../../Src/Amr \
../../Src/EB \
../../Src/Particle \
../../Src/Boundary \
../../Src/Extern \
../../Src/Extern/HYPRE \
../../Src/Extern/hpgmg \ \
../../Src/Extern/HDF5 \
../../Src/Extern/PETSc \
../../Src/Extern/amrdata \
../../Src/Extern/Conduit \
../../Src/Extern/SUNDIALS
../../Src/Extern/ProfParser \
../../Src/Extern/SENSEI \
../../Src/Extern/SWFFT \
../../Src/AmrCore \
../../Src/LinearSolvers \
../../Src/LinearSolvers/MLMG \
../../Src/LinearSolvers/OpenBC \
../../Src/F_Interfaces \
../../Src/F_Interfaces/Octree \
../../Src/F_Interfaces/Base \
../../Src/F_Interfaces/Particle \
../../Src/F_Interfaces/AmrCore \
../../Src/F_Interfaces/LinearSolvers
```

Comment (ax3l) [reply Docs/Doxygen/doxygen.conf]: I didn't add it yet because I saw no futher warnings, but why not! :)

Comment (ax3l) [reply Docs/Doxygen/doxygen.conf]: Thanks for the list, updated accordingly :)

We excluded F_Interfaces/ so far, so I kept it that way for now.

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**